pkg/users/users.js

```diff
@@ -99,11 +99,14 @@ function AccountsPage() {
         return [handleUtmp, handleShadow, handleLogindef];
     }, [], null, handles => handles.forEach(handle => handle.close()));
 
-    // lastlog uses same sorting as /etc/passwd therefore arrays can be combined based on index
     const accountsInfo = useMemo(() => {
         if (accounts && details)
-            return accounts.map((account, i) => {
-                return Object.assign({}, account, details[i]);
+            return accounts.map((account) => {
+                const detail = details.find(detail => detail.name === account.name);
+                if (detail)
+                    return Object.assign({}, account, detail);
+
+                return account;
```

Comment:
In my PR we have a failing tests due to a race condition which this commit resolves: as on account creation `/etc/passwd` and `/etc/shadow` race so if passwd is parsed first then we would end up with the wrong state as lastlog was not parsed yet. Could you move this fix into a separate commit with a description of the race? I'm also happy to that if you don't mind.

See: https://github.com/cockpit-project/cockpit/pull/20087#issuecomment-1963725010

I'm also seeing if I can verify the lastlog2 changes in this PR the `trim()` seems suspicious as we already trim either spaces or tabs.

Reply:
Yup, let's get this merged with a commit for the race and a commit for the `lastlog2` issue maybe with a comment to the util-linux issue showing why we added the `trim()`.